### PR TITLE
Doctest default flags

### LINKED ISSUE #2471: The default doctest flags are hard-coded, they should be configurable
In `sphinx.ext.doctest`, the default doctest flags are hard-coded:

```
        self.opt = doctest.DONT_ACCEPT_TRUE_FOR_1 | doctest.ELLIPSIS | \
            doctest.IGNORE_EXCEPTION_DETAIL
```

They should be put in a configuration variable

```
    app.add_config_value('doctest_default_flags', ...)
```

and used accordingly:

```
        self.opt = self.config.doctest_default_flags
```

### PR DESCRIPTION
**Pull request by taschini**
This addresses and should close #2471.

### CODE REVIEW

File: doc/ext/doctest.rst [hunk 1]
```diff
@@ -176,6 +173,16 @@ The following is an example for the usage of the directives.  The test via
 
 There are also these config values for customizing the doctest extension:
 
+.. confval:: doctest_default_flags
+
+   By default, these options are enabled: ``ELLIPSIS`` (allowing you
```
Comment (lehmannro) [inline doc/ext/doctest.rst]: I'd make this a list, because it is.

File: doc/ext/doctest.rst [hunk 2]
```diff
@@ -55,13 +55,10 @@ names.
...
 .. rst:directive:: .. doctest:: [group]
 
-   A doctest-style code block.  You can use standard :mod:`doctest` flags for
-   controlling how actual output is compared with what you give as output.  By
-   default, these options are enabled: ``ELLIPSIS`` (allowing you to put
-   ellipses in the expected output that match anything in the actual output),
-   ``IGNORE_EXCEPTION_DETAIL`` (not comparing tracebacks),
-   ``DONT_ACCEPT_TRUE_FOR_1`` (by default, doctest accepts "True" in the output
-   where "1" is given -- this is a relic of pre-Python 2.2 times).
+   A doctest-style code block.  You can use standard :mod:`doctest`
```
Comment (lehmannro) [inline doc/ext/doctest.rst]: Here (and below) you rewrapped to something below 80 characters.  All other documents use 80 characters width.  Care to remain consistent?

File: doc/ext/doctest.rst [hunk 3]
```diff
@@ -176,6 +173,16 @@ The following is an example for the usage of the directives.  The test via
...
 
+.. confval:: doctest_default_flags
+
+   By default, these options are enabled: ``ELLIPSIS`` (allowing you
+   to put ellipses in the expected output that match anything in the
+   actual output), ``IGNORE_EXCEPTION_DETAIL`` (everything following
+   the leftmost colon and any module information in the exception name
+   is ignored), ``DONT_ACCEPT_TRUE_FOR_1`` (by default, doctest
+   accepts "True" in the output where "1" is given -- this is a relic
+   of pre-Python 2.2 times).
```
Comment (lehmannro) [inline doc/ext/doctest.rst]: Should have a `.. versionadded::`, or a `versionchanged` above.

**Comment by lehmannro** [comment]
Except for the documentation nits, LGTM.

**Comment by taschini** [comment]
I incorporated the feedback on the doc.

File: doc/ext/doctest.rst (hunk 1 above)
Comment (taschini) [inline doc/ext/doctest.rst]: So would I — or even better,  a set. Still, in doctest it isn't, and I'd rather keep it consistent with that.

Comment (lehmannro) [inline doc/ext/doctest.rst]: Sorry, I was referring to the _markup_:

``` restructuredtext
By default, these options are enabled:
- ``ELLIPSIS``, allowing you to...
- ``IGNORE_EXCEPTION_DETAIL``, causing everything following the leftmost...
- ...
```

Comment (taschini) [inline doc/ext/doctest.rst]: Sorry, I completely misunderstood your comment. I'll do that.

**Comment by taschini** [comment]
Incorporated the last item of feedback from @lehmannro.

**Comment by tk0miya** [comment]
LGTM!
Are there any more comments? I will merge this within a day.

**Comment by tk0miya** [comment]
Merged.
Thank you for contirbution!

**Pull request merged**